readbuffer.go

```diff
@@ -6,21 +6,25 @@ import (
 	"io"
 	"sync"
 	"time"
+
+	"github.com/sirupsen/logrus"
 )
 
 const (
-	MaxBuffer = 1 << 20
+	MaxBuffer = 1 << 21
```

Comment:
How is this number defined? Can maxBuffer be dynamically increased if it can't hold more data? same like the array size that is dynamically increased when buffer exceeded...

Reply:
Talked offline, this is a flow control issue. Quote from darren:

> it's a flow control issue.  If the tunnel client is sending data at 1mb/s and the user is reading data at 1kb/s we will just fill up memory